edb/edgeql/compiler/setgen.py

```diff
@@ -1401,7 +1409,10 @@ def newctx() -> Iterator[context.ContextLevel]:
             remapped_source = new_set_from_set(
                 rptr.source, rptr=rptr.source.rptr,
                 preserve_scope_ns=True, ctx=ctx)
-            subctx.view_map[inner_path_id] = remapped_source
+            key = inner_path_id.strip_namespace(inner_path_id.namespace)
+            assert key not in subctx.view_map
+            subctx.view_map[key] = (inner_path_id, remapped_source)
```

Comment:
Hmm. I wonder if using bare `path_id` here is safe.  Is it possible to construct a case where this would get overwritten incorrectly?  Sibling `WITH` bindings maybe?

Reply:
It looks like `view_map` is always *empty*, (created from the qlctx view_map), so that shouldn't be an issue